cas-server-support-osf/src/main/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction.java

```diff
@@ -336,6 +338,14 @@ protected OpenScienceFrameworkCredential constructCredential(final RequestContex
             // Build and return the credential
             credential.setUsername(remoteUserInfo.getUsername());
             credential.setInstitutionId(remoteUserInfo.getInstitutionId());
+            if (StringUtils.isEmpty(remoteUser)) {
+                logger.error(
```

Comment:
The error message is postponed to here when username and institution ID have been received.